### PR TITLE
Create deploy.yml

### PR DESCRIPTION
**Pull request by runephilosof-abtion**
For trusted publishing on RubyGems

https://guides.rubygems.org/trusted-publishing/adding-a-publisher/

### CODE REVIEW

**Comment by jeppester** [comment]
I have rebased to fix the bundle audit problem

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      # Set up
+      - uses: actions/checkout@v4
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+          ruby-version: ruby
+
+      # Release
+      - uses: rubygems/release-gem@v1
```
Comment (jeppester) [inline .github/workflows/deploy.yml:29]: Does this mean that new versions of the gem will be automatically pushed to rubygems?

**Review by jeppester — APPROVED** [review]
This is the first time I've heard of "trusted publishing" on rubygems, so I cannot review it in details, but it looks good to me :shrug:

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (runephilosof-abtion) [reply .github/workflows/deploy.yml:29]: It is going to call `rake release` https://github.com/rubygems/release-gem/blob/v1/action.yml#L30C12-L30C36

**Comment by jeppester** [comment]
For this to work we'd have to also set up "gem-release":
https://bundler.io/guides/creating_gem.html#releasing-the-gem

I'm closing this because I don't think anyone is using `abt` and so it does not make a lot of sense to spent time on maintaining it.

**Pull request closed without merging**